pytorch_lightning/accelerators/accelerator.py

```diff
@@ -202,6 +202,21 @@ def sync_tensor(self,
         """
         raise NotImplementedError()
 
+    @property
+    def optimizer_states(self) -> List[dict]:
+        """
+        Returns states of optimizers. Allows for syncing/collating optimizer state from processes in custom
+        plugins.
+        Return:
+            Optimizer state dicts
+        """
+        if self.ddp_plugin:
+            return self.ddp_plugin.optimizer_states(self.trainer)
+        optimizer_states = []
+        for i, optimizer in enumerate(self.trainer.optimizers):
+            optimizer_states.append(optimizer.state_dict())
+        return optimizer_states
```

Comment:
this part is now also repeated both inside the accelerator and ddp plugin - i'm wondering how we can avoid the repetition

Reply:
I refactored the code to just override the state_dict getter, still dupes the `state_dict()` call but I think that's better than the loop of before!